src/pushjack/apns.py

```diff
@@ -764,6 +776,346 @@ class APNSExpiredToken(namedtuple('APNSExpiredToken', ['token', 'timestamp'])):
     pass
 
 
+class HttpStatus(object):
+    OK = 200
+    BAD_REQUEST = 400
+    FORBIDDEN = 403
+    METHOD_NOT_ALLOWED = 405
+    GONE = 410
+    PAYLOAD_TOO_LARGE = 413
+    TOO_MANY_REQUESTS = 429
+    INTERNAL_SERVER_ERROR = 500
+    SERVICE_UNAVAILABLE = 503
+
+
+class AuthTokenBase(object):
+    # Sub-class must assign values to them in __init__.
+    # The only reason why these vars added here is to increase the readability.
+
+    # The date and the time the last encryption was performed.
+    encrypted_at = None
+    team_id = None
+    key_id = None
+    algorithm = None
+    # Base64URL-encoded token in JWT format.
+    _encrypted_token = None
+
+    def value(self):
+        raise NotImplementedError(
+            '{} must be implemented'.format(self.__class__.value.__name__))
+
+    @property
+    def encrypted_token(self):
+        if not self._encrypted_token or not self.is_fresh():
+            self._encrypted_token = self._encrypt()
+
+        return self._encrypted_token
+
+    def _encrypt(self):
+        """
+        Encrypt and return the auth token.
+        """
+        self.encrypted_at = _utc_now()
```

Comment:
`_utc_now()` returns a `tz` aware UTC now, it doesn't serialise it.

Reply:
My point was that PyJWT seems to be able to handle datetime conversion to epoch just fine. It looks like it just converts it to utctimetuple() and then converts that to epoch using calendar.timegm() which drops tzinfo anyway.

Reply:
Thanks for the further explanation. 

We are doing `date time` arithmetic in `is_fresh`. I think it would be helpful to use `tz` aware UTC for that purpose, in particular when one operand is created before the time change window and the other is created within the time change window. 

Let me know if I didn't get it right.

Reply:
The datetime calculation is just a difference so it shouldn't matter which base timezone (or lack of) is used. So `naive_dt_now - naive_dt_last` should have the same results as `utc_dt_now - utc_dt_last`. And since we have no need for the timezone in the calculation, it would be better not to introduce another project dependency (pytz isn't currently required by this project).

I would also be fine with just using `time.time()` directly since this is what PyJWT ultimately converts the `iat` value to: https://github.com/jpadilla/pyjwt/blob/ed28e495f937f50165a252fd5696a82942cd83a7/jwt/api_jwt.py#L53-L56

And just to verify:

```python
>>> from datetime import datetime
>>> import time
>>> import calendar
>>> int(time.time()) == calendar.timegm(datetime.utcnow().utctimetuple())
True
>>> (int(time.time()), calendar.timegm(datetime.utcnow().utctimetuple()))
(1553615539, 1553615539)
```